src/ansys/fluent/core/quantity.py

```diff
@@ -1,1369 +1,869 @@
+                )
 
-    Returns
-    -------
-    si_unit: str
-        SI unit
-    """
-    dims_to_float = [float(dim) for dim in dim_list]
-    dim_list = dims_to_float + ((9 - len(dims_to_float)) * [0.0])
-    si_unit = ""
-    dim_to_unit_map = UnitSystem("SI").base_units()
-
-    for key, power in zip(_UnitsTable.dimension_order.values(), dim_list):
-        unit_str = dim_to_unit_map[key]
-        spacechar = " " if len(si_unit) > 0 else ""
-        if power == 1.0:
-            si_unit += spacechar + unit_str
-        elif power != 0.0:
-            si_unit += (
-                spacechar
-                + unit_str
-                + "^"
-                + str(int(power) if power.is_integer() else power)
-            )
-    return si_unit
-
-
-class UnitSystem:
-    """Unit systems."""
-
-    _dim_to_unit_sys_map = {
-        "SI": {
-            "M": "kg",
-            "L": "m",
-            "T": "s",
-            "Temp": "K",
-            "I": "A",
-            "N": "mol",
-            "Iv": "cd",
-            "Angle": "radian",
-            "SAngle": "sr",
-        },
-        "CGS": {
-            "M": "g",
-            "L": "cm",
-            "T": "s",
-            "Temp": "K",
-            "I": "A",
-            "N": "mol",
-            "Iv": "cd",
-            "Angle": "radian",
-            "SAngle": "sr",
-        },
-        "BT": {
-            "M": "slug",
-            "L": "ft",
-            "T": "s",
-            "Temp": "R",
-            "I": "A",
-            "N": "slugmol",
-            "Iv": "cd",
-            "Angle": "radian",
-            "SAngle": "sr",
-        },
-    }
-
-    _supported_unit_sys = ("SI", "CGS", "BT")
-
-    def __init__(self, unit_sys):
-        """
-        Basic unit systems.
+            # Retrieve derived unit composition unit string and factor.
+            if unit_term in self._derived_units:
+                si_multiplier *= (
+                    self._derived_units[unit_term]["factor"] ** unit_term_power
+                )
 
-        Parameters
-        ----------
-        unit_sys: str
-            System name
-        """
-        self._unit_system = unit_sys.upper()
+                # Recursively parse composition unit string
+                si_unit_str, si_multiplier, si_offset = self.si_data(
+                    unit_str=self._derived_units[unit_term]["composition"],
+                    power=unit_term_power,
+                    si_unit_str=si_unit_str,
+                    si_multiplier=si_multiplier,
+                    si_offset=si_offset,
+                )
 
-        if self._unit_system not in UnitSystem._supported_unit_sys:
-            raise ValueError(
-                "Unsupported unit system, only 'SI', 'CGS', 'BT' is allowed."
-            )
+        return self.condense(si_unit_str), si_multiplier, si_offset
 
-    def _get_unit_from_dim(self, dim_list: list) -> str:
-        """
-        Get unit from dimensions.
+    def si_conversion(
```

Comment:
`si_conversion()` is also not using anything from the `UnitsTable` object. Maybe it would be better to rework it, or make it a module method rather than a `UnitsTable` method?